src/Function.h

```diff
@@ -59,6 +59,30 @@ class CVQualifiers;
 
 ///////////////////////////////////////////////////////////////////////////////
 
+class FunctionAttribute
+{
+public:
+	string name;
+	int FuncAttrProb;
+	FunctionAttribute(){};
+	virtual void output_attr(std::ostream &) = 0;
+};
+
+class BooleanFunctionAttribute : public FunctionAttribute
+{
+public:
+	BooleanFunctionAttribute(int, string);
+	void output_attr(std::ostream &);
+};
+
+class VisibilityFunctionAttribute : public FunctionAttribute
```

Comment:
This can be probably generalized into `MultivalueFunctionAttribute` which can be handy for next attributes.

Reply:
You mean removing `VisibilityFunctionAttribute`, `UsedFunctionAttribute`, `NoSanitizeFunctionAttribute` and creating a `MultivalueFunctionAttribute`

Reply:
Yep.